File: LeetCode/268. Missing Number.cpp
// Author : @AdarshGaur

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int x = 0, n = nums.size();
        for(int i=0; i<= n; i++)
            x ^= i;
        for(int &i: nums)
            x ^= i;
        return x;
    }
};

File: LeetCode/268. Missing Number.java
class Solution {
    public int missingNumber(int[] nums) {
        int xor = nums.length;
        for(int i=0; i<nums.length; i++){
            xor ^= i;
            xor ^= nums[i];
        }
        return xor;
    }

//     public int missingNumber(int[] nums) {
//         long sum = 0L;
//         for(int val: nums){
//             sum = sum + val;
//         }
//         int len = nums.length;
//         return (int)(((len * (len + 1L)) >> 1) - sum);
//     }
}

